extensions/ql-vscode/src/run-queries.ts

```diff
@@ -98,6 +94,16 @@ export class QueryEvaluationInfo {
     return path.join(this.querySaveDir, `sortedResults-${resultSetName}.bqrs`);
   }
 
+  /**
+   * Creates a file in the query directory that indicates when this query was created.
+   * This is important for keeping track of when queries should be removed.
+   */
+  async createTimestampFile() {
```

Comment:
Shouldn't this be part of the information we store for the query in the all-history.json?

Happy to be told that that would be too big of a change right now.

Reply:
It can't be stored there. We need to store this value in the query directory so that all workspaces know how old a given query folder is.

Remember, that the goal is to be able to clean up queries for workspaces that you never intend to open again.

Reply:
> It can't be stored there. We need to store this value in the query directory so that all workspaces know how old a given query folder is.
Is all-history.json not meant to be stored in global storage?

Reply:
That's correct. `all-history.json` is in workspace storage. This way each workspace only sees the queries relevant to itself. Other workspaces' queries will not show.

I can change the name and be more clear.